app-next/components/MealsList/MealsList.jsx

```diff
@@ -0,0 +1,33 @@
+"use client"
+import { useEffect, useState } from "react";
+import Meal from "../Meal/Meal";
+import styles from "./MealsList.module.css";
+import api from "@/utils/api";
+
+export const MealsList = () => {
+    const [meals, setMeals] = useState([]);
+
+    useEffect(() => {
+        async function fetchMeals() {
+            const result = await fetch(api("/meals"));
+            const jsonResult = await result.json();
+            setMeals(jsonResult);
+        }
+        fetchMeals();
+    }, []);
+
+    return (
+        <div>
+            <h2>Meals</h2>
+            {meals.length === 0 ? (
+                <p>Loading...</p>
+            ) : (
+                <div className={styles.grid}>
+                    {meals.map((meal) => (
+                        <Meal key={meal.id} meal={meal} />
+                    ))}
+                </div>
+            )}
+        </div>
+    );
```

Comment:
Suggestion: using just a normal if-else statement to define what is returned by the component. This is way clearer and easier to read
```suggestion
if(meals.length === 0) { 
return (
                <p>Loading...</p>
            )}
    return (
        <div>
            <h2>Meals</h2>
                <div className={styles.grid}>
                    {meals.map((meal) => (
                        <Meal key={meal.id} meal={meal} />
                    ))}
                </div>
        </div>
    );
```